rest/server_context.go

```diff
@@ -1200,21 +1203,112 @@ var tempConnectionDetailsForManagementEndpoints = func() (serverAddress string,
 	return base.UnitTestUrl(), base.TestClusterUsername(), base.TestClusterPassword(), "", "", ""
 }
 
-func (sc *ServerContext) ObtainManagementEndpoints() ([]string, error) {
+func (sc *ServerContext) ObtainManagementEndpointsAndHTTPClient() ([]string, *http.Client, error) {
 	clusterAddress, clusterUser, clusterPass, certPath, keyPath, caCertPath := tempConnectionDetailsForManagementEndpoints()
 	agent, err := initClusterAgent(clusterAddress, clusterUser, clusterPass, certPath, keyPath, caCertPath, sc.config.ServerReadTimeout)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	managementEndpoints := agent.MgmtEps()
 
+	httpClient := &http.Client{
+		Transport: agent.HTTPClient().Transport,
+	}
+
 	err = agent.Close()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	return managementEndpoints, httpClient, nil
+}
+
+func (sc *ServerContext) CheckPermissions(httpClient *http.Client, managementEndpoints []string, username, password string, requestedPermissions []string) (statusCode int, permissionResults map[string]bool, err error) {
+	body := []byte(strings.Join(requestedPermissions, ","))
+	statusCode, bodyResponse, err := doHTTPAuthRequest(httpClient, username, password, "POST", "/pools/default/checkPermissions", managementEndpoints, body)
+	if err != nil {
+		return http.StatusInternalServerError, nil, err
+	}
+
+	if statusCode != http.StatusOK {
+		if statusCode == http.StatusUnauthorized {
+			return http.StatusUnauthorized, nil, nil
+		}
+
+		// If we don't provide permissions we get a BadRequest but know we have successfully authenticated
+		if statusCode == http.StatusBadRequest && len(requestedPermissions) > 0 {
+			return statusCode, nil, nil
+		}
+	}
+
+	// At this point we know the user exists, now check whether they have the required permissions
+	if len(requestedPermissions) > 0 {
+		var permissions map[string]bool
+
+		err = base.JSONUnmarshal(bodyResponse, &permissions)
+		if err != nil {
+			return http.StatusInternalServerError, nil, err
+		}
+
+		for _, permResult := range permissions {
+			if permResult {
+				return http.StatusOK, permissions, nil
+			}
+		}
+	}
+
+	return http.StatusForbidden, permissionResults, nil
+}
+
+func doHTTPAuthRequest(httpClient *http.Client, username, password, method, path string, endpoints []string, requestBody []byte) (statusCode int, responseBody []byte, err error) {
+	var httpResponse *http.Response
+
+	maxRetryCount := 5
+	retryCount := 0
+
+	for {
+		endpointIdx := retryCount % len(endpoints)
+		req, err := http.NewRequest(method, endpoints[endpointIdx]+path, bytes.NewBuffer(requestBody))
+		if err != nil {
+			return 0, nil, err
+		}
+
+		req.SetBasicAuth(username, password)
+
+		httpResponse, err = httpClient.Do(req)
+
+		// If no error on request break out and continue
+		if err == nil {
+			break
+		}
```

Comment:
Is there a reason why we're not using the normal retry loop helpers?

Reply:
Guess not, can look into that

Reply:
Done